File: DSA 1/1.c
#include<stdio.h>
#include<stdlib.h>
int * targ(int arr[],int target){
    int* oparr = (int*)malloc(2 * sizeof(int));
    for(int i=0;i<5;i++){
        for(int j=i;j<5;j++){
            if(i!=j){
                if((arr[i]+ arr[j])==target){
                    oparr[0]=arr[i];
                    oparr[1]=arr[j];
                    return oparr;
                }
            }
            
        }
    }
    return NULL;
}

int main(){
    int arr[5]={1,3,5,2,6};
    int target;
    printf("Enter target: ");
    scanf("%d",&target);
    int * updatedArr=targ(arr,target);
    if(updatedArr!=NULL){
        for(int i=0;i<2;i++){
        printf("%d\t",updatedArr[i]);
    }
    }else{
        printf("No Sum found.");
    }
    
    return 0;
}

File: DSA_3/1.js
let arr = [-1, 4, -1, 5, 9, 0, -6, 2];
let target = 44;
let len = arr.length;
let sum = 0;

if (len >= 3) {
  let min = Math.abs(arr[0] + arr[1] + arr[2] - target); 

  for (let i = 0; i < len; i++) {
    for (let j = 0; j < len; j++) {
      for (let k = 0; k < len; k++) {
        if (i !== j && j !== k && k !== i) {
          if (Math.abs(arr[i] + arr[j] + arr[k] - target) <= min) { 
            sum = arr[i] + arr[j] + arr[k];
            min = Math.abs(arr[i] + arr[j] + arr[k] - target); 
          }
        }
      }
    }
  }
}

console.log(sum);
